docker.sh

```diff
@@ -33,21 +36,30 @@ if [ "${DEPEND}" == "build" ]; then
     echo "# To just pull latest/develop version (default) run"
     echo "# DEPEND=pull $0"
     ${DEBUG} docker build \
-        --tag pecan/depends:latest \
-        --file docker/base/Dockerfile.depends .
+        --build-arg R_VERSION=${R_VERSION} \
+        --tag pecan/depends:${IMAGE_VERSION} \
+        docker/depends
 elif [ "${DEPEND}" == "pull" ]; then
     echo "# docker image for dependencies is not build by default."
     echo "# this image takes a long time to build. To build this"
     echo "# image run DEPEND=build $0"
     if [ "${PECAN_GIT_BRANCH}" != "master" ]; then
-      echo "# this will pull develop of base image and tag as latest"
-      echo "# To disable run DEPEND=nothing $0"
-      ${DEBUG} docker pull pecan/depends:develop
-      ${DEBUG} docker tag pecan/depends:develop pecan/depends:latest
+        echo "# this will pull develop of base image and tag as latest"
+        echo "# To disable run DEPEND=nothing $0"
+        if [ "$( docker image ls -q pecan/depends:develop )" == "" ]; then
+            ${DEBUG} docker pull pecan/depends:develop
+        fi
+        ${DEBUG} docker tag pecan/depends:develop pecan/depends:${IMAGE_VERSION}
     else
-      echo "# this will pull latest of base image"
-      echo "# To disable run DEPEND=nothing $0"
-      ${DEBUG} docker pull pecan/depends:latest
+        echo "# this will pull latest of base image"
+        echo "# To disable run DEPEND=nothing $0"
+        if [ "$( docker image ls -q pecan/depends:latest )" == "" ]; then
+            ${DEBUG} docker pull pecan/depends:latest
+        fi
+        if [ "${IMAGE_VERSION}" != "latest" ]; then
+            ${DEBUG} docker tag pecan/depends:latest pecan/depends:${IMAGE_VERSION}
+            ${DEBUG} docker image rm pecan/depends:latest
+        fi
```

Comment:
I think this tagging step should also happen in the `else` block below (if `DEPEND` is unset).